src/main/scala/redis/api/Streams.scala

```diff
@@ -0,0 +1,126 @@
+  extends SimpleClusterKey[K] with RedisCommandMultiBulk[Seq[StreamEntry[F, V]]] {
+  val isMasterOnly = false
+  val encodedRequest: ByteString = encode("XRANGE", Xrange.buildArgs(key, start, end, count))
+  def decodeReply(reply: MultiBulk): Seq[StreamEntry[F, V]] = StreamEntryDecoder.toSeqEntry(reply)
+}
+
+private [redis] object Xrange {
+  def buildArgs[K](key: K, id1: RequestStreamId, id2: RequestStreamId, count: Option[Long])(implicit serializerK: ByteStringSerializer[K]): Seq[ByteString] = {
+    val builder = Seq.newBuilder[ByteString]
+    builder += serializerK.serialize(key)
+    builder += id1.serialize
+    builder += id2.serialize
+    if (count.isDefined) {
+      builder += ByteString("COUNT")
+      builder += ByteString(count.get.toString)
+    }
+    builder.result()
+  }
+}
+
+case class Xrevrange[K, F, V](key: K, end: RequestStreamId, start: RequestStreamId, count: Option[Long] = None)(implicit serializerK: ByteStringSerializer[K], deserializerF: ByteStringDeserializer[F], deserializerV: ByteStringDeserializer[V])
+  extends SimpleClusterKey[K] with RedisCommandMultiBulk[Seq[StreamEntry[F, V]]] {
+  val isMasterOnly = false
+  val encodedRequest: ByteString = encode("XREVRANGE", Xrange.buildArgs(key, end, start, count))
+  def decodeReply(reply: MultiBulk): Seq[StreamEntry[F, V]] = StreamEntryDecoder.toSeqEntry(reply)
+}
+
+case class Xtrim[K](key: K, trimStrategy: TrimStrategy)(implicit serializerK: ByteStringSerializer[K])
+  extends SimpleClusterKey[K] with RedisCommandIntegerLong {
+  val isMasterOnly = true
+  val encodedRequest: ByteString = encode("XTRIM", Seq(keyAsString) ++ trimStrategy.toByteString)
+}
+
+case class Xread[K, F, V](streams: Seq[(K, RequestStreamId)], count: Option[Long])(implicit serializerK: ByteStringSerializer[K], deserializerK: ByteStringDeserializer[K], deserializerF: ByteStringDeserializer[F], deserializerV: ByteStringDeserializer[V])
+  extends RedisCommandMultiBulk[Option[Seq[(K, Seq[StreamEntry[F, V]])]]] {
+  val keys = streams.map(_._1)
+  val isMasterOnly = false
+  val encodedRequest: ByteString = {
+    val builder = Seq.newBuilder[ByteString]
+
+    if (count.isDefined) {
+      builder += ByteString("COUNT")
+      builder += ByteString(count.get.toString)
+    }
+
+    builder += ByteString("STREAMS")
+    builder ++= streams.map(p => serializerK.serialize(p._1))
+    builder ++= streams.map(p => p._2.serialize)
+
+    encode("XREAD", builder.result())
+  }
+
+  def decodeReply(r: MultiBulk): Option[Seq[(K, Seq[StreamEntry[F, V]])]] =
+    StreamEntryDecoder.toOptionSeqStringSeqEntry(r)(deserializerK, deserializerF, deserializerV)
+}
+
+private [redis] object StreamEntryDecoder {
+  def toEntry[F, V](mb: MultiBulk)(implicit deserializerF: ByteStringDeserializer[F], deserializerV: ByteStringDeserializer[V]): StreamEntry[F, V] = {
+    val r = mb.responses.get
```

Comment:
You are using "unsafe" method here and there (`.get`, r(1) (seq.apply(index))
I don't know how safe it is to do it here.
Maybe you could try to compare with what we did in this file https://github.com/etaty/rediscala/blob/55573df26adbe20be54291ffc67414af028bf889/src/main/scala/redis/Converter.scala#L13

Reply:
Please refer to the comment above for the reasoning behind usage of "unsafe" methods.